Algorithm/src/BatchRun.cpp

```diff
@@ -0,0 +1,163 @@
+#include "BatchRun.hpp"
+#include <iostream>
+#include <unistd.h>
+#include <fstream>
+#include <sstream>
+#include <vector>
+#include <string>
+#include <toml++/toml.hpp>
+#include <toml++/toml.h>
+#include "ParseParams.hpp"
+#include "ParamsCarrier.hpp"
+#include <cstdlib>
+#include <filesystem>
+#include "GeliosphereAlgorithm.hpp"
+
+
+using namespace std;
+
+std::vector<std::string> splitString(const std::string &s, char delimiter) {
+    std::vector<std::string> tokens;
+    std::istringstream tokenStream(s);
+    std::string token;
+    while (std::getline(tokenStream, token, delimiter)) {
+        tokens.push_back(token);
+    }
+    return tokens;
+}
+
+std::vector<std::vector<std::string>> readFile(ParamsCarrier *singleTone){
```

Comment:
I do not think, we need to implement separate functions for reading CSV files. In project, we already have dependency on [RapidCSV](https://github.com/d99kris/rapidcsv) library. Please, try to utilize it.